var/spack/repos/builtin/packages/axom/package.py

```diff
@@ -107,14 +107,14 @@ class Axom(CachedCMakePackage, CudaPackage):
     depends_on("raja+openmp", when="+raja+openmp")
     depends_on("raja+cuda", when="+raja+cuda")
 
-    depends_on("umpire@5.0.1~openmp", when="+umpire~openmp")
-    depends_on("umpire@5.0.1+openmp", when="+umpire+openmp")
-    depends_on("umpire@5.0.1+cuda", when="+umpire+cuda")
+    depends_on("umpire@5.0.1:5~openmp", when="+umpire~openmp")
+    depends_on("umpire@5.0.1:5+openmp", when="+umpire+openmp")
+    depends_on("umpire@5.0.1:5+cuda", when="+umpire+cuda")
 
     for sm_ in CudaPackage.cuda_arch_values:
         depends_on('raja cuda_arch={0}'.format(sm_),
                    when='+raja cuda_arch={0}'.format(sm_))
-        depends_on('umpire@5.0.1 cuda_arch={0}'.format(sm_),
+        depends_on('umpire@5.0.1:5 cuda_arch={0}'.format(sm_),
```

Comment:
Sorry for being to nit-picky, but you can just remove the version constraint here, since it's already covered above.

```suggestion
        depends_on('umpire cuda_arch={0}'.format(sm_),
```

In fact above you can simplify to

```
with when('+umpire'):
    depends_on('umpire@5.0.1:5')
    depends_on('umpire +openmp', when='+openmp')
    depends_on('umpire +cuda', when='+cuda')
```

If you don't repeat constraints you're less likely to forget about them when refactoring later.